### DISCUSSION TITLE
LDAP Support

### DISCUSSION THREAD

**Discussion by Tintifax74** [SaraSmiseth/rpi-prosody#46] [Ideas]
Hello

I do appreciate your work providing prosody on docker on RPI. i'm running this setup now for a few years to provide a "private chatserver" to my family. Group of interested additional users has grown since then. to be able "easy" manage them i thought about using LDAP for user/group/auth administration.  Is there as chance to "enrich" this docker image with built-on LDAP support for prosody?

cheers,
Tintifax

p.s. i'm "non-techie". i tried to install auth_ldap on my own inside the container just to give it a try, but then lualdap is requested which seems to need some ldap libraries in the OS inside the container ... i need years to get into this and it seems to be wrong way anyways as i presume it is better to "enhance" the dockerfile with required LDAP items and built a new docker image than "fumbling around" inside container.

**Comment by SaraSmiseth** [comment]
Hey Tintifax!

Thanks for reaching out, and I'm glad to hear that the Docker image has been helpful for your private chat server.

Regarding your question about LDAP support for prosody, it's not something thats currently built-in, but I'm definitely looking into it! Adding the necessary LDAP libraries and modules to the Dockerfile during the image build process seems like a good solution, and I'll prepare a test image with LDAP support soon.

Thanks for your feedback, and feel free to reach out if you have any other questions or concerns.

**Reply by SaraSmiseth** [reply]
Hey there,

Just wanted to give you an update on the LDAP support for the Prosody Docker image. I've made the necessary changes, which you can check out at https://github.com/SaraSmiseth/prosody/commit/81e9c1abd95f798160726f6140afe6527ef7589e.

I have to admit, I haven't tested it myself as I'm not very familiar with LDAP. But I did add some new environment variables that start with LDAP_ and an AUTHENTICATION setting that can be set to "ldap". You can find all the details in the updated readme at https://github.com/SaraSmiseth/prosody/blob/dev/readme.md#environment-variables. Essentially, these are the same settings you can find here: https://modules.prosody.im/mod_auth_ldap.

These changes are now available in the sarasmiseth/prosody:edge image. So if you have time, feel free to test it out and let me know how it goes!

Thanks for your interest in the project, and please don't hesitate to reach out if you have any questions.

Cheers,
Sara

**Reply by Tintifax74** [reply]
Superb. I'll give it a try soon and will let u know.

in fact i have 2 more comments, but not LDAP related:
1) add turn/coturn support --> i'm running my prosody behind NAT. using jabber/xmpp to setup calls works in current config only if both clients are in same network. i managed to "inject" that somehow into running container and can provide u with prosody config value for that.
2) http upload size --> i think by default it is set to 1mb which for pictures is enough from my PoV in most cases. but for videos it is too less and android client scales down video to hit 1mb size-limit which leads to a "bunch of large pixels" 

shall i open 2 new "ideas" for that?

**Reply by SaraSmiseth** [reply]
Yes, please open two new "ideas" for those features. And please share your configuration or changes for the turn/coturn support, that would be a huge help to get me started.
